## Hypothesis Testing

In [11]:
import numpy as np
import pandas as pd
from scipy.stats import f_oneway
from scipy.stats import chi2_contingency
from scipy.stats import ttest_ind

df = pd.read_csv('D:\internship\impactsenseAI-Infosys-Intern-project\data\preprocessed_earthquake_data.csv')

#### 1. Magnitude Difference by Earthquake Type
Hypothesis: The mean earthquake magnitude differs significantly between reviewed and non-reviewed events.

In [12]:
# Select relevant groups, drop missing values
group1 = df[df['status'] == 'reviewed']['mag'].dropna()
group2 = df[df['status'] != 'reviewed']['mag'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: -1.0748565111189015
p-value: 0.28494966597262317


**Statement**:The t-test result for Hypothesis-1 yields a t-statistic of -1.07 and a p-value of 0.285. Since the p-value is greater than the common significance threshold of 0.05, there is not enough evidence to reject the null hypothesis. Thus, the mean earthquake magnitude does not differ significantly between reviewed and non-reviewed events in your data.

#### 2. Depth Difference by Magnitude Type
Hypothesis: The mean depth of earthquakes significantly varies between 'mb' and 'mw' magnitude type categories.

In [13]:
group1 = df[df['magType'] == 'mb']['depth'].dropna()
group2 = df[df['magType'] == 'mw']['depth'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 22.514758657461467
p-value: 8.933021714439036e-112


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 22.51 and a p-value of approximately $8.93 \times 10^{-112}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is extremely strong evidence to reject the null hypothesis. Therefore, the mean depth of earthquakes is significantly different between `mb` and `mw` magnitude type events in your dataset.

#### 3. Magnitude Consistency Across Months
Hypothesis: There is no significant association between the month of occurrence and the categorical bins of earthquake magnitude (such as low, medium, high).

In [14]:
# Bin magnitudes into categories
df['mag_bin'] = pd.cut(df['mag'], bins=[0, 3.5, 5.0, 6.5, 9], labels=['Low', 'Medium', 'High', 'Very High'])

# Contingency table: mag_bin vs Month
contingency = pd.crosstab(df['mag_bin'], df['Month'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 46.542726462177434
Degrees of freedom: 33
p-value: 0.05922370884750974


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 46.54, degrees of freedom of 33, and a p-value of approximately 0.0592. Since this p-value is slightly above the conventional significance threshold of 0.05, there is insufficient evidence to reject the null hypothesis at the 5% level. Therefore, the data does not provide strong enough evidence to conclude that earthquake magnitude category and month of occurrence are dependent variables.

#### 4. DayOfWeek Consistency Across Event Type
Hypothesis: The type of event ('type' column: earthquake, quarry blast, etc.) is independent of the day of week
(DayOfWeek) on which it occurred.

In [15]:
contingency = pd.crosstab(df['type'], df['DayOfWeek'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 92.16992465854213
Degrees of freedom: 36
p-value: 8.07286408381138e-07


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 92.17, degrees of freedom of 36, and a p-value of approximately $8.072 \times 10^{-07}$
 . Since this p-value is far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the day of the week and event type are significantly dependent, indicating a meaningful association between these categorical variables in your data.

#### 5. Magnitude Mean Difference by Year Decade
Hypothesis: The mean earthquake magnitude (mag) differs significantly between records from two different decades (e.g., 2000s vs 2010s).

In [16]:
df['decade'] = (df['Year'] // 10) * 10
dec1 = df[df['decade'] == 2000]['mag'].dropna()
dec2 = df[df['decade'] == 2010]['mag'].dropna()

ttest_result = ttest_ind(dec1, dec2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 7.034358911416218
p-value: 2.0392846020245177e-12


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 7.03 and a p-value of approximately $2.039 \times 10^{-12}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the mean earthquake magnitude differs significantly between the two compared decades in your dataset.

### Task:

- Suggest five additional hypothesis tests, based on the preprocessed earthquake dataset, that cover both numerical and categorical variables using t-tests and chi-square tests, to extend the depth of your analysis.

#### 6. Regional Recurrence Hypothesis
Hypothesis: The frequency of earthquakes is significantly associated with specific latitude-longitude regions — meaning some regions experience earthquakes far more often than others.

In [17]:
# Bin latitude and longitude into categories
df['lat_bin'] = pd.cut(df['latitude'], bins=[-90, -30, 0, 30, 90],
labels=['South', 'Subtropical South', 'Subtropical North', 'North'])
df['lon_bin'] = pd.cut(df['longitude'], bins=[-180, -120, -60, 0, 60, 120, 180],
labels=['Far West', 'West', 'Central', 'East', 'Far East', 'Extreme East'])

# Combine to create region categories
df['region'] = df['lat_bin'].astype(str) + '_' + df['lon_bin'].astype(str)

# Contingency table: region vs event type
contingency = pd.crosstab(df['region'], df['type'])

# Chi-square test
chi2, p, dof, expected = chi2_contingency(contingency)

print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)


Chi-square statistic: 52554.27500417052
Degrees of freedom: 138
p-value: 0.0


**Statement**: The chi-square test yields a statistic of 52554.28 with 138 degrees of freedom and a p-value near 0.0. Since the p-value is far below 0.05, we reject the null hypothesis. Thus, earthquake occurrences significantly depend on geographic region, showing clear regional clustering.

#### 7. Depth Variation by Earthquake Type Hypothesis

Hypothesis: The mean earthquake depth differs significantly among different earthquake types, suggesting that certain types of earthquakes tend to occur deeper within the Earth’s crust than others.

In [18]:
# Group data by earthquake type
groups = [group['depth'].values for name, group in df.groupby('type')]
# Perform one-way ANOVA
f_stat, p_val = f_oneway(*groups)

print(f"F-statistic: {f_stat}")
print(f"p-value: {p_val}")

F-statistic: 26.416900502337423
p-value: 1.2982931307086624e-31


**Statement**: The one-way ANOVA test yields an F-statistic of 26.42 with a p-value of approximately $1.29 \times 10^{-31}$. Since the p-value is far below 0.05, we reject the null hypothesis. This indicates that earthquake depth significantly varies by earthquake type, implying that some types originate much deeper than others.

#### 8. Temporal Frequency Hypothesis

Hypothesis: The mean number of earthquakes differs significantly between the periods 1900–1950 and 1951–2000, suggesting that earthquake activity has changed over time.

In [19]:
# Count earthquakes per year
yearly_counts = df.groupby('Year').size().reset_index(name='count')

# Split data into two decades (example: 1900–1950 vs 1951–2000)
decade1 = yearly_counts[yearly_counts['Year'].between(1900, 1950)]['count']
decade2 = yearly_counts[yearly_counts['Year'].between(1951, 2000)]['count']

# Perform independent t-test
t_stat, p_val = ttest_ind(decade1, decade2, equal_var=False)

print(f"T-statistic: {t_stat}")
print(f"P-value: {p_val}")

T-statistic: -10.238600001355772
P-value: 5.541469517745341e-14
